assets/app/scripts/services/docker.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("DockerImageGeneratorService", function($filter, ApplicationGenerator, DataService) {
+    var findImage = function(name, projectContext) {
+      var importImage = {
+        kind: "ImageStreamImport",
+        apiVersion: "v1",
+        metadata: {
+          name: "newapp",
```

Comment:
> doesn't necessarily mean its right ;-) if it does cause an error, then arguably name should be optional if import is false (probably easier said than done) cc @deads2k

My memory of the code was complaining that it was sticky.  As I recall, @smarterclayton uses the name to locate which secrets he's allowed to use (and maybe other information about insecure-import) during the ISI task *before* he decides if he's actually going to import of not.

Now, overloading the name to mean that, maybe not ideal.  Regardless, we're not likely rip it out for 1.2 and that means we support it forever.